src/Functional/SortBy.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright (C) 2011-2017 by Lars Strojny <lstrojny@php.net>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+namespace Functional;
+
+use Traversable;
+
+/**
+ * Sorts a collection by a value of some function of its elements
+ *
+ * @param Traversable|array $collection
+ * @param callable $getProperty function($element, $collection) returns int
+ * @param bool $preserveKeys
+ * @see sort()
+ * @return array
+ */
+function sort_by($collection, callable $getProperty, $preserveKeys = false)
+{
+    return sort(
+        $collection,
+        function($left, $right, $collection) use ($getProperty) {
+            return $getProperty($left, $collection) - $getProperty($right, $collection);
```

Comment:
Shouldn’t it use spaceship here for correctness?

Basically `$getProperty(…) <=> $getProperty(…)`?